code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/notification/builders/AEPPushNotificationBuilder.kt

```diff
@@ -0,0 +1,977 @@
+            Log.warning(
+                PushTemplateConstants.LOG_TAG,
+                SELF_TAG,
+                "Package manager NameNotFoundException while reading default application icon: ${e.localizedMessage}"
+            )
+        }
+        return -1
+    }
+
+    /**
+     * Sets a custom color to the notification's small icon.
+     *
+     * @param builder the notification builder
+     * @param iconColorHex `String` containing a color code to be used in customizing the
+     * small icon color
+     */
+    private fun setSmallIconColor(
+        builder: NotificationCompat.Builder,
+        iconColorHex: String?
+    ) {
+        if (iconColorHex.isNullOrEmpty()) {
+            Log.trace(
+                PushTemplateConstants.LOG_TAG,
+                SELF_TAG,
+                "Empty icon color hex string found, custom color will not be applied to the notification icon."
+            )
+            return
+        }
+
+        try {
+            // sets the icon color if provided
+            builder.setColorized(true).color = Color.parseColor("#$iconColorHex")
+        } catch (exception: IllegalArgumentException) {
+            Log.trace(
+                PushTemplateConstants.LOG_TAG,
+                SELF_TAG,
+                "Unrecognized hex string passed to Color.parseColor(), custom color will not be applied to the notification icon."
+            )
+        }
+    }
+
+    /**
+     * Returns the resource id for the drawable with the given name. The file must be in the
+     * res/drawable directory. If the drawable file is not found, 0 is returned.
+     *
+     * @param iconName the name of the icon file
+     * @param context the application [Context]
+     * @return the resource id for the icon with the given name
+     */
+    private fun getIconWithResourceName(
+        iconName: String?,
+        context: Context
+    ): Int {
+        return if (iconName.isNullOrEmpty()) {
+            0
+        } else context.resources.getIdentifier(iconName, "drawable", context.packageName)
+    }
+
+    /**
+     * Checks if the icon is valid.
+     *
+     * @param icon the icon to be checked
+     * @return true if the icon is valid, false otherwise
+     */
+    private fun isValidIcon(icon: Int): Boolean {
+        return icon > 0
+    }
+
+    /**
+     * Creates a notification channel if the device is running on Android O or higher. If the channel
+     * already exists, the same channel is used. A default channel ID will be used if no channel ID
+     * is received from the payload.
+     *
+     * @param context the application [Context]
+     * @param channelId [String] containing the notification channel id
+     * @param customSound `String` containing the custom sound to apply on the notification channel
+     * @param importance [Int] containing the notification importance
+     * @return A [String] containing the created or existing channel ID
+     */
+    internal fun createChannel(
```

Comment:
Can you update the function name to something like createChannelIfNeeded / createChannelIfRequired?